jbmc/src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -2347,33 +2363,30 @@ codet java_bytecode_convert_methodt::convert_instructions(
       results[0]=
         binary_predicate_exprt(op[0], ID_java_instanceof, arg0);
     }
-    else if(statement=="monitorenter")
-    {
-      // becomes a function call
-      code_typet type;
-      type.return_type()=void_typet();
-      type.parameters().resize(1);
-      type.parameters()[0].type()=java_reference_type(void_typet());
-      code_function_callt call;
-      call.function()=symbol_exprt("java::monitorenter", type);
-      call.lhs().make_nil();
-      call.arguments().push_back(op[0]);
-      call.add_source_location()=i_it->source_location;
-      c=call;
-    }
-    else if(statement=="monitorexit")
+    else if(statement == "monitorenter" || statement == "monitorexit")
     {
+      std::string descriptor;
```

Comment:
Nit pick: you can safely make this an `irep_idt`, though admittedly it doesn't make much of a difference.